src/cli/Vaults.ts

```diff
@@ -29,7 +30,36 @@ function makeListVaultsCommand() {
     );
 }
 
-function makeAddVaultCommand() {
+function makeScanVaultsCommand() {
+  return new commander.Command('scan')
+    .description('scan a known peer for accessible vaults')
+    .option('--node-path <nodePath>', 'node path')
+    .requiredOption('-p, --public-key <publicKey>', 'name of vault')
+    .option('-v, --verbose', 'increase verbosity by one level')
+    .arguments('name of vault to remove')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const nodePath = determineNodePath(options);
+        const verbose: boolean = options.verbose ?? false;
+
+        const publicKeyPath = options.publicKey;
+        const publicKey = fs.readFileSync(publicKeyPath);
+
+        const vaultNames = await client.scanVaultNames(nodePath, publicKey.toString());
```

Comment:
This is a perfect situation to create models that take the client, using an IoC container to instantiate those objects and using those.

However I realized due to the short-term nature of the `pk` command, it doesn't make sense to use IoC in the same as the way its done for servers and for `pk-agent`.

Instead the IoC container for the `pk` should instead be lazy. So it should only construct and then cache what is needed. It should be possible to do this by wrapping the container's access with lambdas (arrow functions).